### DISCUSSION TITLE
Error with Manubot generating citations from sources.yaml

### DISCUSSION THREAD

**Discussion by mrittika-dey** [greenelab/lab-website-template#211] [Q&A] [answered]
Hello!
I recently spotted this error when I tried to update citations from my sources.yaml file:
ImportError: cannot import name 'RequestRate' from 'pyrate_limiter' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/pyrate_limiter/__init__.py)

I thought it might be due to the new version of pyrate_limiter which does not seem to have any module called 'RequestRate'. I rolled back the pyrate_limiter version being used to v2.1.0, which seemed to work. Now, I have a new error which says:
AttributeError: 'Limiter' object has no attribute 'ratelimit'

Both of these errors occur when calling the function "get_pubmed_ids_for_doi". 
It would be great if someone could tell me how I can solve this issue.

**Comment by vincerubinetti** [comment, marked as answer]
See https://github.com/greenelab/lab-website-template/issues/210

For a quick fix, ~update your Manubot version in `/_cite/requirements.txt` to `0.5.6` or later.~ see above.

I'm not sure what effect just updating the sub-dependencies would have; it would probably be difficult or impossible to get it to work properly. This was an upstream issue in Manubot which they've fixed, so upgrading just Manubot should work.

**Reply by mrittika-dey** [reply]
Hi Vincent! 
I tried that before and just tried again, I still have the "AttributeError: 'Limiter' object has no attribute 'ratelimit'" error. I followed the other comments on the pull request concerning this, let us know what happens!

**Reply by vincerubinetti** [reply]
Sorry yes, upgrading to Manubot 0.5.6 won't actually work, I was mistaken. See the linked issue for a temporary fix that should actually work.

**Reply by vincerubinetti** [reply]
Manubot 0.6.0 is out on PYPI, fixing the issue. LWT 1.1.6 is out with a change to the python dependency versions.
